programs/steward/src/state/steward_state.rs

```diff
@@ -447,7 +471,26 @@ impl StewardState {
         self.delegations[self.num_pool_validators] = Delegation::default();
         self.instant_unstake.set(self.num_pool_validators, false)?;
         self.progress.set(self.num_pool_validators, false)?;
+        self.progress.set(self.num_pool_validators, false)?;
+
+        Ok(())
+    }
 
+    /// Mark a validator for removal from the pool - this happens right after
+    /// `remove_validator_from_pool` has been called on the stake pool
+    /// This is cleaned up in the next epoch
+    pub fn mark_validator_for_removal(&mut self, index: usize) -> Result<()> {
+        self.validators_to_remove.set(index, true)?;
+        Ok(())
+    }
+
+    /// Called when adding a validator to the pool so that we can ensure a 1-1 mapping between
+    /// the validator list and the steward state
+    pub fn increment_validator_to_add(&mut self) -> Result<()> {
+        self.validators_added = self
+            .validators_added
+            .checked_add(1)
+            .ok_or(StewardError::ArithmeticError)?;
         Ok(())
```

Comment:
nit: can just be the statement without the Ok(())

Reply:
Done